sdk/core/Azure.Core.Experimental/src/Variant/Value.cs

```diff
@@ -0,0 +1,1347 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Azure {
+    /// <summary>
+    /// Used to store primitive values without boxing, and other intstances.
```

Comment:
```suggestion
    /// Used to store primitive values without boxing, and other instances.
```

Reply:
Would be nice to write a couple sentences here on the approach and how it works.

Reply:
I will improve up the comments and docs in a subsequent PR once the implementation/APIs stabilize.